packages/amplify_datastore_plugin_interface/lib/src/types/models/auth_mode_strategy.dart

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+/// Represents different auth strategies supported by a client
+/// interfacing with an AppSync backend
+enum AuthModeStrategy {
+  /// Default authorization type read from API configuration
+  default$,
+
+  /// Uses schema metadata to create a prioritized list of potential
+  /// authorization types that could be used for a request. The client iterates
+  /// through that list until one of the available types succeeds or all of
+  /// them fail.
+  multiAuth,
+}
+
+/// Helpers for [AuthModeStrategy].
+extension AuthModeStrategyX on AuthModeStrategy {
+  /// The raw value used for interfacing with native SDKs.
+  String get rawValue {
+    switch (this) {
+      case AuthModeStrategy.default$:
```

Comment:
It looks like that the codebase mostly uses `describeEnum` as convention. But using `describeEnum` doesn't work with this escaped keyword...
Do we have option to name `default` as `defaultAuth`, or use uppercase like Android?
Or do developers need to use `AuthModeStrategy.default` at any occasion instead of passing `null`?

Reply:
Yeah, since `default` is a reserved word you can't use it as a member name. `defaultAuth` is fine, but it seemed redundant to say auth again. 

Personally, I don't like using `null` in this context if there is a default value hidden behind it.

Reply:
I might make this `AuthModeStrategy.defaultValue` or `AuthModeStrategy.defaultStrategy` even though it is redundant.

My brain read this as `AuthModeStrategy.$default` and I was wondering if this was generated.